### PR TITLE
Migration/gracenotegroup

### PR DESCRIPTION
**Pull request by rvilarl**
Quick win! :)

### CODE REVIEW

File: src/gracenotegroup.ts [hunk 1]
```diff
@@ -14,19 +14,40 @@ import { Beam } from './beam';
 import { StaveTie } from './stavetie';
 import { TabTie } from './tabtie';
 import { StaveNote } from './stavenote';
+import { Note } from './note';
 
-// To enable logging for this class. Set `Vex.Flow.GraceNoteGroup.DEBUG` to `true`.
-function L(...args) {
+export interface GraceNoteGroupFormatOptions {
```
Comment (wassertim) [inline src/gracenotegroup.ts:19]: if it is for internal use only, should it be exported?

File: src/gracenotegroup.ts [hunk 2]
```diff
@@ -90,7 +108,7 @@ export class GraceNoteGroup extends Modifier {
     this.preFormatted = false;
 
     this.show_slur = show_slur;
-    this.slur = null;
+    this.slur = undefined;
```
Comment (wassertim) [inline src/gracenotegroup.ts:111]: changing null to undefined can be a breaking change. I've seen a lot of places with:
```javascript
if (a === null) ...
```
which is not the same as:
```javascript
if (a === undefined) ...
```

File: src/gracenotegroup.ts (hunk 1 above)
Comment (rvilarl) [reply src/gracenotegroup.ts:19]: Hi good to see you! I believe that anyone using the `format` will require this definition.

File: src/gracenotegroup.ts (hunk 2 above)
Comment (rvilarl) [reply src/gracenotegroup.ts:111]: I agree I have already changed 
```
if (a === null)
```
by 
```
if (!a)
```
If we also leave `null` the declarations, parameters and return values will get very complex.

Comment (wassertim) [reply src/gracenotegroup.ts:111]: same problem here. I would not change that:
```javascript
if (a === null)
```
is not the same as:
```javascript
if (!a)
```

Comment (wassertim) [reply src/gracenotegroup.ts:111]: Maybe it was not intentional @0xfe? Then it is a fine change

File: src/gracenotegroup.ts (hunk 1 above)
Comment (0xfe) [reply src/gracenotegroup.ts:19]: I'm not a fan of these single-field structures, it makes it harder to read the code as you jump around through types. Let's just use inline types here.

File: src/gracenotegroup.ts [hunk 3]
```diff
@@ -14,19 +14,40 @@ import { Beam } from './beam';
...
+  protected beams: Beam[];
+
+  static get CATEGORY(): string {
     return 'gracenotegroups';
   }
 
-  // Arrange groups inside a `ModifierContext`
-  static format(gracenote_groups, state) {
+  /** Arranges groups inside a `ModifierContext`. */
+  static format(gracenote_groups: GraceNoteGroup[], state: GraceNoteGroupFormatOptions): boolean {
```
Comment (0xfe) [inline src/gracenotegroup.ts:50]: Prefer: `state: {  left_shift: number }`

File: src/gracenotegroup.ts (hunk 2 above)
Comment (0xfe) [reply src/gracenotegroup.ts:111]: This case is okay (though I agree with @wassertim that we should be careful with these.) `undefined` is semantically more correct here than `null`.

**Comment by rvilarl** [comment]
This one should be now ready for merge.

File: src/gracenotegroup.ts [hunk 4]
```diff
@@ -14,19 +14,33 @@ import { Beam } from './beam';
 import { StaveTie } from './stavetie';
 import { TabTie } from './tabtie';
 import { StaveNote } from './stavenote';
+import { Note } from './note';
 
-// To enable logging for this class. Set `Vex.Flow.GraceNoteGroup.DEBUG` to `true`.
-function L(...args) {
+// To enable logging for this class. Set `GraceNoteGroup.DEBUG` to `true`.
+function L(...args: string[]) {
```
Comment (0xfe) [inline src/gracenotegroup.ts:20]: This should be `any`. It goes right into `console.log`.

Comment (rvilarl) [reply src/gracenotegroup.ts]: ok, done

**Comment by rvilarl** [comment]
Ready to merge.

**Comment by 0xfe** [comment]
Thanks!

**Pull request merged**